go/arrow/array/compare.go

```diff
@@ -114,6 +114,16 @@ func ArrayEqual(left, right Interface) bool {
 	}
 }
 
+// ArraySliceEqual reports whether slices left[lbeg:lend] and right[rbeg:rend] are equal.
+func ArraySliceEqual(left Interface, lbeg, lend int64, right Interface, rbeg, rend int64) bool {
```

Comment:
Why do we use `int64` for slice bounds while we use `int` for array length? Should we use `int` for everything?

Otherwise LGTM.

Reply:
yeah, I think @nickpoorman made that comment at some point.
we probably need to homogenize a bit this corner of the code.

I initially thought `int` would always suffice for arrays as I thought they would always be backed by some kind of `[]T`.
but then I came across `array.Chunked` which may consist of multiple `[]T` presented as a logical `[]T`. (same for `Records`)

so... we'd probably need to mass rename `s/int/int64/`.

would you mind opening a jira ticket about this?

Reply:
Unless we create some sort of "chunks" concept for storing things like arrays of columns, I've been taking the approach that row length int64 and column length is int.

Reply:
For anything that uses `array.Chunked` I've been using `int64` and anything else using the underlying Go arrays I've been using `int`.